doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+CREATE MATERIALIZED VIEW <name> FROM ... [IN CLUSTER <cluster>]
+```
+
+If no cluster is explicitly specified, the active cluster is used. If there
+is no active cluster, the `CREATE VIEW` statement is rejected.
+
+Users cannot read data from a view on which they do not have the `SELECT`
+privilege. Note that if the view references sources or other views, privileges
+on those sources and other views are not considered when determining whether a
+user has access to the view.
+
+### Index
+
+The existing `CREATE INDEX` statement is extended to support specifying the
+cluster in which to create the index:
+
+```
+CREATE INDEX <name> ON <object> (<col>, ...) [IN CLUSTER <cluster>]
+```
+
+If no cluster is explicitly specified, the active cluster is used. If there
+is no active cluster, the `CREATE INDEX` statement is rejected.
+
+`SHOW INDEXES` and `mz_indexes` will gain an additional column specifying to
+which cluster the index belongs.
+
+There are no access control restrictions on individual indexes. Access to
+indexes is controlled by the permissions on the cluster containing the index.
+
+### Secret
+
+A secret is a new type of database object that securely stores a confidential
+credential.
+
+To create a new secret:
+
+```
+CREATE SECRET <name> AS '<secret>';
+```
+
+To drop an existing secret:
+
+```
+DROP SECRET <name>;
+```
+
+The existence of the secret will be stored in the catalog, but the actual
+contents of the secret will be securely stored in a secrets management system.
+
+Sources and sinks will need to learn to accept secret references in places that
+they currently expect an inline string or file path. Designing that new syntax
+is TODO.
+
+Referencing a secret in a source or sink definition requires that the user who
+creates the source or sink has `USAGE` privilege on the secret.
+
+There is intentionally no command to view a secret, though a sufficiently
+motivated attacker with the `USAGE` privilege could likely extract a secret with
+a well-constructed `CREATE SOURCE` command that references the secret.
+
+### Timeline
+
+TODO: this section is still relatively poorly understood. Are timelines 1:1
+with databases, or their own separate entity?
+
+TODO: engage @mjibson.
+
+TODO: add timelines to the overview diagram.
+
+UX thoughts from @frankmcsherry on timelines:
+
+> Within each REGION there may be multiple TIMELINEs, which correspond roughly
+> to coordinator timelines. A TIMELINE contains collections, and is the unit
+> within which it is possible to query across multiple collections.
+>
+> The main restrictions we plan to impose on users is that their access to data
+> is always associated with a REGION, CLUSTER, and TIMELINE, and attempts to
+> access data across these may result in poorer performance, or explicit query
+> rejection.
+
+### Role
+
+A role assigns a set of privileges to a group of users. Roles are scoped to a
+single region.
+
+Privileges can be granted on databases, schemas, and most database objects.
```

Comment:
what about "types" of objects? Like there is the notion of "this user can access running clusters but doesn't have the ability to start new clusters because they only get to access named resources we already pay for, not cause the bill to increase in any way". How would one grant/deny the privileges for that?

Reply:
Good point. It's not something that PostgreSQL has any opinions on, I don't think. Snowflake has an explicit `CREATE <TYPE>` permission (e.g., `CREATE WAREHOUSE`). We might want to do the same for `CREATE CLUSTER`!

Reply:
Added to the doc.